apps/api/src/teams/event-registrations/event-registrations.service.ts

```diff
@@ -1,38 +1,65 @@
 import type { FilterQuery } from '@mikro-orm/core';
 import { InjectRepository } from '@mikro-orm/nestjs';
-import { BadRequestException, ForbiddenException, Injectable } from '@nestjs/common';
+import {
+  BadRequestException,
+  ForbiddenException,
+  Injectable,
+} from '@nestjs/common';
 import { BaseRepository } from '../../shared/lib/orm/base.repository';
 import { Role } from '../../shared/modules/authorization/types/role.enum';
-import type { PaginatedResult, PaginateDto } from '../../shared/modules/pagination';
+import type {
+  PaginatedResult,
+  PaginateDto,
+} from '../../shared/modules/pagination';
 import type { User } from '../../users/user.entity';
 import { TeamEvent } from '../events/team-event.entity';
 import { TeamMember } from '../members/team-member.entity';
+import type { CreateTeamEventRegistrationDTO } from './dto/create-team-event-registration.dto';
 import type { ListRegisteredEventsDto } from './dto/list-registered-events.dto';
 import { TeamEventRegistration } from './team-event-registration.entity';
 
 @Injectable()
 export class TeamEventRegistrationsService {
   constructor(
-    @InjectRepository(TeamMember) private readonly teamMemberRepository: BaseRepository<TeamMember>,
-    @InjectRepository(TeamEvent) private readonly teamEventRepository: BaseRepository<TeamEvent>,
+    @InjectRepository(TeamMember)
+    private readonly teamMemberRepository: BaseRepository<TeamMember>,
+    @InjectRepository(TeamEvent)
+    private readonly teamEventRepository: BaseRepository<TeamEvent>,
     @InjectRepository(TeamEventRegistration)
     private readonly teamEventRegistrationRepository: BaseRepository<TeamEventRegistration>,
   ) {}
 
-  public async create(user: User, teamEventId: number): Promise<TeamEventRegistration> {
-    const event = await this.teamEventRepository.findOneOrFail({ teamEventId }, { populate: ['createdBy', 'team'] });
+  public async create(
+    user: User,
+    teamEventId: number,
+    createTeamEventRegistrationDTO: CreateTeamEventRegistrationDTO,
+  ): Promise<TeamEventRegistration> {
+    const event = await this.teamEventRepository.findOneOrFail(
+      { teamEventId },
+      { populate: ['createdBy', 'team'] },
+    );
 
     if (event.private) {
-      const membership = await this.teamMemberRepository.findOne({ user, team: { teamId: event.team.teamId } });
+      const membership = await this.teamMemberRepository.findOne({
+        user,
+        team: { teamId: event.team.teamId },
+      });
       if (!membership)
-        throw new ForbiddenException('Not a team member');
+throw new ForbiddenException('Not a team member');
```

Comment:
Pourquoi l'avoir changé
```suggestion
        throw new ForbiddenException('Not a team member');
```